rust/agents/relayer/src/msg/metadata/ccip_read.rs

```diff
@@ -0,0 +1,100 @@
+use async_trait::async_trait;
+use hyperlane_ethereum::OffchainLookup;
+use reqwest::Client;
+use serde::{Deserialize, Serialize};
+use serde_json::json;
+use std::ops::Deref;
+
+use derive_new::new;
+use eyre::Context;
+use tracing::{info, instrument};
+
+use super::{BaseMetadataBuilder, MetadataBuilder};
+use ethers::abi::AbiDecode;
+use ethers::core::utils::hex::decode as hex_decode;
+use hyperlane_core::{HyperlaneMessage, H256};
+use regex::Regex;
+
+#[derive(Serialize, Deserialize)]
+struct OffchainResponse {
+    data: String,
+}
+
+#[derive(Clone, Debug, new)]
+pub struct CcipReadIsmMetadataBuilder {
+    base: BaseMetadataBuilder,
+}
+
+impl Deref for CcipReadIsmMetadataBuilder {
+    type Target = BaseMetadataBuilder;
+
+    fn deref(&self) -> &Self::Target {
+        &self.base
+    }
+}
+
+#[async_trait]
+impl MetadataBuilder for CcipReadIsmMetadataBuilder {
+    #[instrument(err, skip(self))]
+    async fn build(
+        &self,
+        ism_address: H256,
+        message: &HyperlaneMessage,
+    ) -> eyre::Result<Option<Vec<u8>>> {
+        const CTX: &str = "When fetching CcipRead metadata";
+        let ism = self.build_ccip_read_ism(ism_address).await.context(CTX)?;
+
+        let response = ism.get_offchain_verify_info(message.body.clone()).await;
+        let info: OffchainLookup = match response {
+            Ok(_) => {
+                info!("incorrectly configured getOffchainVerifyInfo, expected revert");
+                return Ok(None);
+            }
+            Err(raw_error) => {
```

Comment:
Probably should have send this to you earlier, but have you seen https://github.com/ensdomains/ethers-ccip-read. I wonder if we could reuse logic from there (looks possible?)

I think @yorhodes and I had chatted that the requirement in the spec to require the `data` field in the JSON response is not great, but services like CCTP actually are already compliant https://developers.circle.com/stablecoin/reference/getattestation

Reply:
(Might have to fork it and have it return the calldata of the final call)

Reply:
For posterity, it seemed a little complicated to use [ethers-ccip-read](https://github.com/ensdomains/ethers-ccip-read) in the relayer. The types used internally don't line up terribly well, and for a simple GET/POST call I also didn't think it's worth bringing in an extra dep

Reply:
I think it's a bit more complicated than implemented here
eg supporting recursive offchain lookups
happy to punt for now but this feels like a good place to minimize surface area